client-apis/encrypt.md

```diff
@@ -113,68 +161,168 @@ This document refers to the output of the key derivation algorithm as the derive
 - [IV Length](../data-format/message-header.md#iv-length): MUST match the [IV length](../framework/algorithm-suites.md#iv-length)
   specified by the [algorithm suite](../framework/algorithm-suites.md)
-- [Frame Length](../data-format/message-header.md#frame-length): MUST be the same value as the input [frame length](#frame-length),
-  if included.
-- [IV](../data-format/message-header.md#iv): MUST have a value of 0, padded to the [IV length](../data-format/message-header.md#iv-length).
-- [Authentication Tag](../data-format/message-header.md#authentication-tag): MUST be the output of the
-  [authenticated encryption algorithm](../framework/algorithm-suites.md#encryption-algorithm)
-  specified by the [algorithm suite](../framework/algorithm-suites.md), with the following inputs:
-  - The AAD is the serialized [message header body](../data-format/message-header.md#header-body)
-  - The IV is the [IV specified above](../data-format/message-header.md#iv)
-  - The cipherkey is the derived data key
-  - The plaintext is an empty byte array
+- [Frame Length](../data-format/message-header.md#frame-length): MUST be the value of the frame size determined above.
 
-Each frame of the [message body](../data-format/message-body.md) is serialized with the following specifics:
+After serializing the message header body,
+this operation MUST calculate an [authentication tag](../data-format/message-header.md#authentication-tag)
+over the message header body.
+The value of this MUST be the output of the [authenticated encryption algorithm](../framework/algorithm-suites.md#encryption-algorithm)
+specified by the [algorithm suite](../framework/algorithm-suites.md), with the following inputs:
 
-- [IV](../data-format/message-body.md#iv): MUST be the [sequence number](../data-format/message-body-aad.md#sequence-number)
-  used in the [message body AAD](../data-format/message-body-aad.md) for this frame.
-- [Encrypted Content](../data-format/message-body.md#encrypted-content): MUST be the output of the [authenticated encryption algorithm](../framework/algorithm-suites.md#encryption-algorithm)
-  specified by the [algorithm suite](../framework/algorithm-suites.md), with the following inputs:
-  - The AAD is the serialized [message body AAD](../data-format/message-body-aad.md)
-  - The IV is the [IV](../data-format/message-body.md#iv) specified for this frame above.
-  - The cipherkey is the derived data key
-  - The plaintext contains part of the input [plaintext](#plaintext) this frame is encrypting.
-- [Authentication Tag](../data-format/message-body.md#authentication-tag): MUST be the authentication tag outputted by the above encryption.
+- The AAD is the serialized [message header body](../data-format/message-header.md#header-body).
+- The IV has a value of 0.
+- The cipherkey is the derived data key
+- The plaintext is an empty byte array
```

Comment:
As earlier, we may want to lift this out to the suites section.

Reply:
If we are thinking of moving this to the suites,
(which I like)
should we also move
the equivalent body parts?
The Body AAD et. al.